llmfoundry/data/finetuning/tasks.py

```diff
@@ -108,6 +155,22 @@ def _tokenize_formatted_example(
     return tokenizer(text=prompt, text_target=response)
 
 
+def _tokenize_formatted_example(
+        example: Example,
+        tokenizer: PreTrainedTokenizerBase) -> TokenizedExample:
+    example_format = _get_example_type(example)
+
+    if example_format == 'chat':
+        chat_example: ChatFormattedDict = example  # type: ignore
```

Comment:
is it possible to get rid of the type ignore and use an assert here instead? 
```
assert isinstance(example, ChatFormattedDict)
return _tokenize...(example, tokenizer)
```

Reply:
sadly not, because isinstance only works for classes not type synonyms